### DISCUSSION TITLE
Welcome to bmai Discussions!

### DISCUSSION THREAD

**Discussion by pappde** [pappde/bmai#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by danlangford** [comment]
I am Dan. I often go by “bagels”. I love programming. My day job is as a software dev team lead at your typical enterprise co. My day job is all java and most of my side work is java, python, JavaScript, etc. I love learning new tech. I have a little experience with C++. I have experience with reverse engineering other code like the Heroes of the Storm replay files.  I don’t have much experience with AI type coding and thought this could be a place for me to get started.

**Comment by pappde** [comment]
BMAI is an example of the type of AI you might use for a board game or traditional classic game. There is a discrete search space, and you want to pick the move that gives you the best expected value, whether by simulating out the entire game to the conclusion or by a heuristic function to judge the goodness of a final position in your search space. It is similar in structure to a computer poker AI that I developed, though poker  is mechanically a simpler game so the focus there was actually on opponent modeling. 

These heuristic functions are an opportunity to work in personality, intelligence, opponent modeling, and learning. For example, an early example of machine learning was used in Backgammon, where the final result of Win or Loss was used to adjust values of the moves that were taken to reach that conclusion.

I have worked on many video game AIs as well - shooters and strategy games. The approach there is generally quite different, though you could actually apply some of the same concepts at a meta-game level. However, a key element of a good AI these more open games is still good heuristic functions.
